File: client-api/src/main/java/org/ostelco/topup/api/auth/AccessTokenPrincipal.java
package org.ostelco.topup.api.auth;

import java.security.Principal;

/**
 * Holds the 'user-id' obtained by verifying and decoding an OAuth2
 * 'access-token'.
 */
public class AccessTokenPrincipal implements Principal {

    private final String subject;

    public AccessTokenPrincipal(String subject) {
        this.subject = subject;
    }

    @Override
    public String getName() {
        return subject;
    }
}

File: prime-modules/src/main/kotlin/org/ostelco/prime/auth/AccessTokenPrincipal.kt
package org.ostelco.prime.auth

import org.ostelco.prime.model.Identity
import java.security.Principal

/**
 * Holds the 'identity' obtained by verifying and decoding an OAuth2 'access-token'.
 */
class AccessTokenPrincipal(val identity: Identity) : Principal {
    override fun getName(): String = identity.id
}
